hicommonwealth/commonwealth
# #6005 - Create Drawer Component

The work in this PR creates a new drawer component. 

## Link to Issue
Closes: #6005 

## Description of Changes
- creates drawer with ability to expand to fullscreen on desktop/tablet devices
- drawer can expand from top, bottom, right, and left. However, styling has not been generalized to take into account all of these directions. For the purposes of this PR, please review as though the only available direction is from the right. Ticket #6325 created to generalize styling. 

## Test Plan
- navigate to `/components` page
- click on 'Open Drawer' and check that drawer has expected styling and behavior

## Other Considerations
- blocks other tickets, such as #5509
